operational_analysis/methods/eya_gap_analysis.py

```diff
@@ -0,0 +1,208 @@
+        """
+        Produce a waterfall plot showing the progression from the EYA to OA estimates of AEP. 
+        
+        Args:
+            data(:obj:`numpy array`): data to be used to create waterfall plot
+            index(:obj:`list`): List of string values to be used for x-axis labels
+            path(:obj:`string`): Location to save waterfall plot
+            
+        Returns:
+            (None)
+        """
+        
+        # Store data and create a blank series to use for the waterfall
+        trans = pd.DataFrame(data = {'amount': data}, index = index) # Assign gaps to data frame
+        blank = trans.amount.cumsum().shift(1).fillna(0) # Perform cumulative sum on gap values
+    
+        # Get the net total number for the final element in the waterfall
+        total = trans.sum().amount
+        trans.loc["oa_aep"]= total # Add new field to gaps data frame
+        blank.loc["oa_aep"] = total # Add new fiekd to cumulative sum data frame
+    
+        # The steps graphically show the levels as well as used for label placement
+        step = blank.reset_index(drop=True).repeat(3).shift(-1)
+        step[1::3] = np.nan
+    
+        # When plotting the last element, we want to show the full bar,
+        # Set the blank to 0
+        blank.loc["oa_aep"] = 0
+    
+        #Plot and label
+        my_plot = trans.plot(kind='bar', stacked=True, bottom=blank, legend=None, 
+                             figsize=(12, 6))
+        my_plot.plot(step.index, step.values,'k')
+        my_plot.set_ylabel("Energy (GWh/yr)")
+    
+        #Get the y-axis position for the labels
+        y_height = trans.amount.cumsum().shift(1).fillna(0)
+    
+        # Get an offset so labels don't sit right on top of the bar
+        mx = trans.max() # Max value in gap analysis values
+        neg_offset = mx / 25
+        pos_offset = mx / 50
+        
+        # Add labels to each bar
+        loop = 0
+        for index, row in trans.iterrows():
+            # For the last item in the list, we don't want to double count
+            if row['amount'] == total:
+                y = y_height[loop]
+            else:
+                y = y_height[loop] + row['amount']
+                
+            # Determine if we want a neg or pos offset
+            if row['amount'] > 0:
+                y += pos_offset
+            else:
+                y -= neg_offset
+            my_plot.annotate("{:,.0f}".format(row['amount']),(loop,y),ha="center")
+            loop += 1
+    
+        # Adjust y-axis to focus on region of interest
+        plt_min = blank[1:-1].min() # Min value in cumulative sum values
+        plt_max = blank[1:].max() # Min value in cumulative sum values
+        my_plot.set_ylim(0.9 * plt_min, 1.1 * plt_max)#blank.max()+int(plot_offset))
+        
+        #Rotate the labels
+        my_plot.set_xticklabels(trans.index,rotation=0)
+ 
+        # Save figure
+        if save_fig_path != False:
```

Comment:
Save figure only if save_fig_path isn't false. Otherwise it is a string with path name.